dd-java-agent/instrumentation/jax-rs-annotations/src/main/java/datadog/trace/instrumentation/jaxrs/JaxRsAnnotationsInstrumentation.java

```diff
@@ -105,9 +107,20 @@ public static Scope nameSpan(@Advice.Origin final Method method) {
       }
       final String resourceName = resourceNameBuilder.toString().trim();
 
-      final Scope scope = GlobalTracer.get().scopeManager().active();
-      if (scope != null && !resourceName.isEmpty()) {
-        Span span = scope.span();
+      final List<Scope> scopeList = new ArrayList<>();
+      Scope parentScope = GlobalTracer.get().scopeManager().active();
+      if (parentScope == null) {
+        parentScope =
+            GlobalTracer.get()
+                .buildSpan("JAX-RS")
```

Comment:
You should default to naming this with the `operationName` to make sure that, even if `resourceName` is empty, we don't end up with spans named the same for all endpoints. Remember this impacts metrics baselines.

Reply:
Great idea.  Updated.